lint-rules.md

```diff
@@ -0,0 +1,1266 @@
+
+When newline separating inputs to a call, always follow each input with a trailing comma. See examples below.
+
+The assignment operator (`=`) should always be surrounded by whitespace.
+
+The `input:` token should always be on the same line and separated by one space from the opening curly bracket (`{`), likeso: `call foo_task { input:`. Never newline-separate the opening bracket (`{`) from the `input:` token.
+
+If there is a singular input and it can fit on that same line while being equal to or under the 90 character width limit, all tokens should be separated by a single space, likeso: `call foo_task { input: bam = sorted_bam }`. Note that there is no trailing comman here. This could optionally be split into 3 lines, likeso:
+
+```wdl
+call foo_task { input:
+    bam = sorted_bam,
+}
+```
+
+Note the trailing comma.
+
+**Group**: `spacing`
+
+#### Example
+
+Good (note that these examples take advantage of the input simplifying mechanism made available in the [v1.1 specification](https://github.com/openwdl/wdl/blob/main/versions/1.1/SPEC.md#call-statement)):
+
+```wdl
+call md5sum.compute_checksum { input: file = bam }
+
+call samtools.quickcheck { input: bam }
+
+call util.compression_integrity { input:
+    bgzipped_file = bam,
+}
+
+call markdups_post_wf.markdups_post { input:
+    markdups_bam = select_first([
+        markdups.duplicate_marked_bam,
+        "undefined",
+    ]),
+    markdups_bam_index = select_first([
+        markdups.duplicate_marked_bam_index,
+        "undefined",
+    ]),
+    coverage_beds,
+    coverage_labels = parse_input.labels,
+    prefix = post_subsample_prefix + ".MarkDuplicates",
+}
+```
+
+### `snake_case`
+
+Variables, tasks, and workflows should be in lowercase "snake_case".
+
+**Group**: `naming`
+
+### `pascal_case`
+
+Declared structs should be in "PascalCase".
+
+**Group**: `naming`
+
+### `indentation`
+
+Indentation should be 4 spaces and never tab literals.
+
+### `trailing_whitespace`
+
+No whitespace on empty lines. No whitespace at the end of lines.
+
+### `one_empty_line`
+
+At most one empty line in a row. There should never be 2 or more blank lines in a row.
+
+**Group**: `spacing`
+
+### `no_newline_eof`
+
+End the file with a newline.
+
+**Group**: `spacing`
+
+### `comment_whitespace`
+
+Comments on the same line as code should have 2 spaces before the `#` and one space before the comment. Comments on their own line should match the indentation level around them and have one space between the `#` and the comment. Keep in mind that even comments must be kept below the 90 character width limit.
```

Comment:
I have a strong dislike for that comment style. I have to admit, it does _look_ a lot cleaner. But I just find manually aligning comments as I write code to be a huge pain. I'm more concerned with my experience as a developer writing comments than I am with someone else reading my comments. I guess I'm selfish in that regard.

More importantly, I find the first example (following my rules) perfectly readable. Mainly because of the 'double space' in it providing a bit of a break between code and comment.

Followup Q about your example: Are the 2 comments there related? Or are they separate ideas? _If_ those are meant to be one sentence (or one thought), I would argue both examples are wrong. I would write that "multiline spanning comment" like this:
```
input {
    # This is a long comment with a lot going on. I need
    # more space to write it out.
    Int x
    String name
}
```
_If_ they are separate thoughts, why would they be aligned? I guess it's more visually appealing, but it doesn't strike me are more logical or coherent. If the comments are unrelated, they shouldn't be aligned. Maybe "shouldn't" is strong, but they certainly don't _need_ to be.

Reply:
@claymcleod 

> I'm in favor of what the spec says FWIW.

I looked over the spec, and I don't see it making any comment on the style issue me and Andrew are discussing. In my reading, it just outlines how to make a comment/how a comment should be parsed. It does not clarify how to style comments.

Maybe I'm looking in the wrong spot. Can you link or quote?